metadata-io/src/main/java/com/linkedin/metadata/entity/ebean/EbeanEntityService.java

```diff
@@ -201,44 +236,82 @@ protected UpdateAspectResult ingestAspectToLocalDB(@Nonnull final Urn urn, @Nonn
       @Nonnull final AuditStamp auditStamp, @Nonnull final SystemMetadata providedSystemMetadata) {
 
     return _entityDao.runInTransactionWithRetry(() -> {
+      final String urnStr = urn.toString();
+      final EbeanAspectV2 latest = _entityDao.getLatestAspect(urnStr, aspectName);
+      long nextVersion = _entityDao.getNextVersion(urnStr, aspectName);
 
-      // 1. Fetch the latest existing version of the aspect.
-      final EbeanAspectV2 latest = _entityDao.getLatestAspect(urn.toString(), aspectName);
-      final EbeanAspectV2 keyAspect = _entityDao.getLatestAspect(urn.toString(), getKeyAspectName(urn));
+      return ingestAspectToLocalDBNoTransaction(urn, aspectName, updateLambda, auditStamp, providedSystemMetadata, latest, nextVersion);
```

Comment:
What is the purpose of calling this "NoTransaction" -- Isn't this running in a transaction loop? (runInTransactionWithRetry)

Reply:
This is just to point out that the inner method is not the one forcing the transaction?

Reply:
yes this method is not running the transaction. Other methods calling this would do this. Couldn't think of a better name so kept it like this